### PR TITLE
Add object filter ratio

### PR DESCRIPTION
**Pull request by mathmaniac43**
Per issue #2948 (take two, this time targeting the correct branch `release-0.11.0`)

I was not able to figure out how to run frigate in a dev. container on a recorded video the way I wanted to for testing, but am open to doing so if I can get a bit of help/instruction.

I also did my best to maintain existing style/phrasing throughout, but feel free to make adjustments/requests if you think it should be improved.

Thanks!

### CODE REVIEW

**Comment by NickM-27** [comment]
Since you added a field to be model you will also need to add a migration class to the migrations folder. Should be able to go off of what's there, let me know if you have any questions on that.

File: frigate/config.py [hunk 1]
```diff
@@ -210,6 +210,14 @@ class FilterConfig(FrigateBaseModel):
     max_area: int = Field(
         default=24000000, title="Maximum area of bounding box for object to be counted."
     )
+    min_ratio: float = Field(
+        default=0,
+        title="Minimum ratio of bounding box's width/height for object to be counted.",
+    )
+    max_ratio: float = Field(
+        default=24000000,  # @todo: Is there a way to make this `math.inf` but still load from YAML?
```
Comment (blakeblackshear) [inline frigate/config.py:218]: I think this is a fine default. Objects can't be less than 1px tall and nobody is going to have a camera that is 24000000  pixels wide.

File: frigate/video.py [hunk 2]
```diff
@@ -38,6 +38,9 @@
 
 def filtered(obj, objects_to_track, object_filters):
     object_name = obj[0]
+    object_score = obj[1]
+    object_bounds = obj[2]
```
Comment (blakeblackshear) [inline frigate/video.py:42]: let's rename this to `object_box` instead of `bounds` for consistency.

File: frigate/video.py [hunk 3]
```diff
@@ -615,8 +626,14 @@ def process_frames(
                 for group in detected_object_groups.values():
 
                     # apply non-maxima suppression to suppress weak, overlapping bounding boxes
+                    bounds = o[2]  # xmin, ymin, xmax, ymax
```
Comment (blakeblackshear) [inline frigate/video.py:629]: Similarly `box` instead of `bounds`.

**Review by blakeblackshear — CHANGES_REQUESTED** [review]

**Comment by blakeblackshear** [comment]
Where are you getting stuck running in the dev container? You should be able to run `python3 -m frigate` in the terminal window.

I use this config for local dev:
```yaml
cameras:
  back:
    ffmpeg:
      inputs:
        - path: /lab/frigate/airport.mp4
          input_args: -re -stream_loop -1 -fflags +genpts
```

**Comment by mathmaniac43** [comment]
@blakeblackshear Python isn't my primary programming language and I haven't developed any packages before, so was not familiar with the `-m` flag, I bet that will get me moving, thanks.

**Comment by mathmaniac43** [comment]
@blakeblackshear I got a `No module named 'frigate.version'` error, which looks like I can resolve using the `Makefile`. However, the dev. container does not have `make` in it, so I am going to have to do it outside the container. May be worth considering adding `make` to the container in the future.

**Comment by NickM-27** [comment]
@mathmaniac43 I believe you are supposed to make outside and then start the container. That's what I've always done

**Comment by blakeblackshear** [comment]
Building outside the container is listed in the docs here: https://docs.frigate.video/contributing#core

The command is now just `make` instead of `make amd64_frigate`. Feel free to fix that in the docs in this PR too.

File: frigate/models.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ class Event(Model):
     box = JSONField()
     area = IntegerField()
     retain_indefinitely = BooleanField(default=False)
+    ratio = FloatField()
```
Comment (NickM-27) [inline frigate/models.py:22]: Now that I am on my computer, just leaving an official comment about the migration being needed.

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/models.py (hunk 4 above)
Comment (mathmaniac43) [reply frigate/models.py:22]: @NickM-27 I added the migration script [here](https://github.com/blakeblackshear/frigate/pull/2952/files/2ad0ed9b8d74475e05c800eecd82bf643db580b5#diff-2546ea6e97f5943ff2ab27f9cfd917da1af7eb19d6cdafd6ca96236bf82b2438), are you saying that any default values should match both there and here?

Comment (NickM-27) [reply frigate/models.py:22]: Sorry didn't see it, a couple notes:
1. yes the default values should be added here as well
2. There is already a 007 migration so this should be updated to 008

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Comment by mathmaniac43** [comment]
@NickM-27 Makes sense, just fixed!

**Review by NickM-27 — APPROVED** [review]

File: frigate/video.py [hunk 5]
```diff
@@ -38,6 +38,9 @@
 
 def filtered(obj, objects_to_track, object_filters):
```
Comment (blakeblackshear) [inline frigate/video.py:39]: You should include the ratio object filter here. This prevents them from being tracked.

File: migrations/008_add_object_filter_ratio.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+"""Peewee migrations -- 007_add_object_filter_ratio.py.
```
Comment (NickM-27) [inline migrations/008_add_object_filter_ratio.py:1]: Need to update this comment with the new number and filename

Comment (NickM-27) [reply migrations/008_add_object_filter_ratio.py:1]: My PR which had a new migration in it just got merged so you'll need to update this file to `009` as well

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Comment by NickM-27** [comment]
Looks good to me👌

**Comment by mathmaniac43** [comment]
@blakeblackshear Pinging to make sure this has not fallen off your radar. Thanks.

**Comment by blakeblackshear** [comment]
Just busy with other things at the moment

**Review by blakeblackshear — APPROVED** [review]

**Pull request merged**